### PR TITLE
Fix upper limit of groupedbar

### PR DESCRIPTION
**Pull request by BeastyBlacksmith**
Fix https://github.com/JuliaPlots/StatsPlots.jl/issues/357

### CODE REVIEW

File: src/bar.jl [hunk 1]
```diff
@@ -48,15 +48,9 @@ grouped_xy(y::AbstractArray) = 1:size(y,1), y
     end
 
     # compute fillrange
-    y, fr = isstack ? groupedbar_fillrange(y) : (y, get(plotattributes, :fillrange, nothing))
```
Comment (daschw) [inline src/bar.jl:51]: So how are the y positions and fillranges for stacked bars calculated now?

Comment (BeastyBlacksmith) [reply src/bar.jl:51]: It's right below:
```julia
fill_bottom = min(minimum(y) / 100, the_ylims[1])
```

Comment (BeastyBlacksmith) [reply src/bar.jl:51]: Happy to have more test cases though

Comment (daschw) [reply src/bar.jl:51]: But they are different for `:stack`ed groupedbars, which is calculated in `groupedbar_fillrange`. These changes only cover the `:dodge` case.

Comment (daschw) [reply src/bar.jl:51]:
```julia
using StatsPlots

x = [
    1 2 2
    3 2 1
]

groupedbar(x, bar_position = :stack)
```
With this PR:
stackedbar

On master:
stackedbar_master

Comment (BeastyBlacksmith) [reply src/bar.jl:51]: should be fixed now

File: src/bar.jl [hunk 2]
```diff
@@ -47,32 +47,34 @@ grouped_xy(y::AbstractArray) = 1:size(y,1), y
...
-function groupedbar_fillrange(y)
+function groupedbar_fillrange(y; y_bottom = 0)
     nr, nc = size(y)
     # bar series fills from y[nr, nc] to fr[nr, nc], y .>= fr
     fr = zeros(nr, nc)
     y = copy(y)
     y[.!isfinite.(y)] .= 0
     for r = 1:nr
-        y_neg = 0
+        y_neg = y_bottom
```
Comment (daschw) [inline src/bar.jl:77]: I don't think this makes sense. This only offsets all negative values by `+y_bottom`. Can't we leave `groupedbar_fillrange` as it was? Isn't this covered by L62 already? Or, respectively, would we even support negative input for log scales?

Comment (BeastyBlacksmith) [reply src/bar.jl:77]: you are right, thats a leftover, I reverted this in https://github.com/JuliaPlots/StatsPlots.jl/commit/bcc35353482e596e32808140ca1257b56164b55c

**Pull request merged**